rules/sortImportsRule.ts

```diff
@@ -8,34 +9,61 @@ export class Rule extends Lint.Rules.AbstractRule {
 }
 
 class SortImportsWalker extends Lint.RuleWalker {
-	public visitSourceFile(node: ts.SourceFile) {
-		for (const importGroup of this.getImportGroups(node)) {
-			const importLines = importGroup.map(line => line.getText(node));
-			const importLinesForComparison = importLines.map(line => line.toLowerCase().replace(' {', ' +'));
-			const sortedImportLines = importLinesForComparison.slice().sort();
-
-			for (let i = 0; i < importLines.length; i += 1) {
-				if (importLinesForComparison[i] !== sortedImportLines[i]) {
+	public visitSourceFile(sf: ts.SourceFile) {
+		for (const importGroup of this.getImportGroups(sf)) {
+			const sortableLinesToImports = new Map<string, ts.ImportDeclaration>();
+			const unsortedLines = importGroup.map(importDeclaration => {
+				// opening brace is below alphanumeric characters char-code-wise, but want named imports above defaults
+				// + comes directly after *, so swap with that
+				const sortableLine = importDeclaration.getText(sf)
+					.toLowerCase()
+					.replace('import {', 'import +');
+				sortableLinesToImports.set(sortableLine, importDeclaration);
+				return sortableLine;
+			});
+
+			const sortedLines = unsortedLines.slice().sort();
+
+			for (let i = 0; i < unsortedLines.length; i += 1) {
```

Comment:
I have a lint rule on that forbids it because += 1 is more explicit and can't be confused with ++i